modular/blocksyncer/blocksyncer_options.go

```diff
@@ -1,478 +0,0 @@
-		if blockSyncerBackup, err := newBackupBlockSyncerService(junoCfg, mainDBIsMaster); err != nil {
-			return nil, err
-		} else {
-			BackupService = blockSyncerBackup
-		}
-	}
-
-	return MainService, nil
-}
-
-// initClient initialize a juno client using given configs
-func (b *BlockSyncerModular) initClient() error {
-	// JunoConfig the runner
-	junoConfig := cmd.NewConfig("juno").
-		WithParseConfig(parsecmdtypes.NewConfig().
-			WithRegistrar(registrar.NewBlockSyncerRegistrar(
-				messages.CosmosMessageAddressesParser,
-			)).WithDBBuilder(db.BlockSyncerDBBuilder).WithFileType("toml"),
-		)
-	cmdCfg := junoConfig.GetParseConfig()
-	cmdCfg.WithTomlConfig(b.config)
-
-	//set toml config to juno config
-	if readErr := parsecmdtypes.ReadConfigPreRunE(cmdCfg)(nil, nil); readErr != nil {
-		log.Infof("readErr: %v", readErr)
-		return readErr
-	}
-
-	// get DSN from env first
-	var dbEnv string
-	if b.Name() == BlockSyncerModularName {
-		dbEnv = DsnBlockSyncer
-	} else {
-		dbEnv = DsnBlockSyncerSwitched
-	}
-
-	dsn, envErr := getDBConfigFromEnv(dbEnv)
-	if envErr != nil {
-		log.Info("failed to get db config from env, use db config from config file")
-		config.Cfg.Database.DSN = b.config.Database.DSN
-	}
-	if dsn != "" {
-		log.Info("use db config from env")
-		config.Cfg.Database.DSN = dsn
-	}
-
-	var ctx *parser.Context
-	ctx, err := parsecmdtypes.GetParserContext(config.Cfg, cmdCfg)
-	if err != nil {
-		log.Errorf("failed to GetParserContext err: %v", err)
-		return err
-	}
-	b.parserCtx = ctx
-	b.parserCtx.Indexer = NewIndexer(ctx.EncodingConfig.Marshaler,
-		ctx.Node,
-		ctx.Database,
-		ctx.Modules,
-		b.Name())
-	return nil
-}
-
-// initDB create tables needed by block syncer. It depends on which modules are configured
-func (b *BlockSyncerModular) initDB(useMigrate bool) error {
-
-	var err error
-	for _, module := range b.parserCtx.Modules {
-		if module, ok := module.(modules.PrepareTablesModule); ok {
-			if useMigrate {
-				err = module.AutoMigrate()
-			} else {
-				err = module.PrepareTables()
-			}
-			if err != nil {
-				log.Errorw("failed to PrepareTables/AutoMigrate tables", "error", err)
-				return err
-			}
-		}
-	}
-	return nil
-}
-
-// serve start BlockSyncer rpc service
-func (b *BlockSyncerModular) serve(ctx context.Context) {
-	migrateDBAny := ctx.Value(MigrateDBKey{})
-	if migrateDB, ok := migrateDBAny.(bool); ok && migrateDB {
-		err := b.initDB(true)
-		if err != nil {
-			log.Errorw("fail to init DB", "error", err)
-			return
-		}
-	}
-	// Create a queue that will collect, aggregate, and export blocks and metadata
-	exportQueue := types.NewQueue(100)
-
-	// Create workers
-	worker := parser.NewWorker(b.parserCtx, exportQueue, 0, config.Cfg.Parser.ConcurrentSync)
-	worker.SetIndexer(b.parserCtx.Indexer)
-
-	latestBlockHeight := mustGetLatestHeight(b.parserCtx)
-	Cast(b.parserCtx.Indexer).GetLatestBlockHeight().Store(int64(latestBlockHeight))
-	go b.getLatestBlockHeight(ctx)
-
-	lastDbBlockHeight := uint64(0)
```

Comment:
To keep a same code style, suggest we wrap the logic here as a function, like `mustGetEpoch` with a retry number.

Reply:
Good advice. I'll change it